pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -0,0 +1,152 @@
+func (f *frontend) preflightValidation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	var header http.Header
+	var b []byte
+
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+	originalPath := r.Context().Value(middleware.ContextKeyOriginalPath).(string)
+	referer := r.Header.Get("Referer")
+
+	subId := chi.URLParam(r, "subscriptionId")
+	resourceGroupName := chi.URLParam(r, "resourceGroupName")
+
+	resources, err := unmarshalRequest(body)
+	if err == nil {
+		resourceCount := len(resources.Resources)
+		ch := make(chan api.ValidationResult, resourceCount)
+		for _, raw := range resources.Resources {
+			// get typeMeta from the raw data
+			typeMeta := &api.ResourceTypeMeta{}
+			err = json.Unmarshal(raw, &typeMeta)
+			if err != nil {
+				// failing to parse the preflight body is not considered a validation failure. continue
+				log.Warningf("bad request. Failed to unmarshal ResourceTypeMeta: %s", err)
+				continue
+			}
+			if strings.EqualFold(typeMeta.Type, "Microsoft.RedHatOpenShift/openShiftClusters") {
+				path := PreflightResourceId(subId, resourceGroupName, typeMeta.Name)
+				ch <- f._preflightValidation(ctx, log, raw, path, originalPath, r.Method, referer, &header, f.apis[typeMeta.APIVersion].OpenShiftClusterConverter, f.apis[typeMeta.APIVersion].OpenShiftClusterStaticValidator)
+			}
+		}
+		close(ch)
+
+		for res := range ch {
+			// go through and return the first error
+			if res.Status == api.ValidationStatusFailed {
+				log.Warningf("preflight validation failed")
+				b, err = marshalValidationResult(&res)
+				reply(log, w, header, b, err)
+				return
+			}
+			b, err = marshalValidationResult(&res)
+		}
+		log.Info("preflight validation succeeded")
+	}
+	reply(log, w, header, b, err)
+}
+
+func (f *frontend) _preflightValidation(ctx context.Context, log *logrus.Entry, raw json.RawMessage, path, originalPath, method, referer string, header *http.Header, converter api.OpenShiftClusterConverter, staticValidator api.OpenShiftClusterStaticValidator) api.ValidationResult {
+	_, err := f.validateSubscriptionState(ctx, path, api.SubscriptionStateRegistered)
+	if err != nil {
+		return api.ValidationResult{
+			Status: api.ValidationStatusFailed,
+			Error: &api.ManagementErrorWithDetails{
+				Message: to.StringPtr(err.Error()),
+			},
+		}
+	}
+
+	// unmarshal raw to OpenShiftCluster type
+	doc := &api.OpenShiftCluster{}
+	if !f.env.IsLocalDevelopmentMode() /* not local dev or CI */ {
+		doc.Properties.FeatureProfile.GatewayEnabled = true
+	}
+
+	var ext = converter.ToExternal(doc)
+	err = json.Unmarshal(raw, &ext)
+	if err != nil {
+		return api.ValidationResult{
+			Status: api.ValidationStatusFailed,
+			Error: &api.ManagementErrorWithDetails{
+				Message: to.StringPtr(err.Error()),
+			},
+		}
+	}
+
+	// For Put operation
```

Comment:
what does this comment mean?

Reply:
Was a old note since we are only looking at cluster create at the moment and should validate other requests later. Removed